src/App.jsx

```diff
@@ -14,33 +15,77 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortfiled, setSortfiled] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const visibileGoods = [...goodsFromServer].sort((a, b) => {
+    if (sortfiled === SORT_BY_ALPHABET) {
+      return a.localeCompare(b);
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (sortfiled === SORT_BY_LENGTH) {
+      return a.length - b.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    visibileGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortfiled === SORT_BY_ALPHABET ? '' : 'is-light'}`}
+          onClick={() => setSortfiled(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${sortfiled === SORT_BY_LENGTH ? '' : 'is-light'}`}
```

Comment:
Ensure that the corrected variable name `sortField` is used here instead of `sortfiled`.